contrib/labstack/echo.v4/echotrace.go

```diff
@@ -47,7 +49,12 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 			defer span.Finish()
 
 			// pass the span through the request context
-			c.SetRequest(request.WithContext(ctx))
+			req := request.WithContext(ctx)
+			c.SetRequest(req)
+			op := httpinstr.StartOperation(httpinstr.MakeHandlerOperationArgs(req, span), nil)
```

Comment:
The problem with your small refactor below is that now the code inside `httpinstr.MakeHandlerOperationArgs` runs every time, adding extra computation even when AppSec is off. Can you fix that somehow please?

Reply:
I did something bold in https://github.com/DataDog/dd-trace-go/pull/1042/commits/01ab93d31dd9894ae438f25eb509ea375fffaf90 which is just adding more `if enabled` guards in `httpinstr` into `StartOperation`, `MakeHandlerOperationArgs` and  `MakeHandlerOperationRes`.

I thought something more advanced with interfaces to lazily build the `HandlerOperationArgs` and `HandlerOperationRes` would be too much for now. I'd rather wait for more integrations before "optimizing" this.

Reply:
What do you think about having something like `appsec.Enabled()` or `appsec.Status() != appsec.StatusDisabled` (the latter being more appropriate in cases where there would be more statuses than on/off). And simply not running the code at all when it's off, instead of trying to make no-op's out of everything. This no-op pattern doesn't seem to work well in your case.

Reply:
Agreed, much better.
Now fixed.